src/app/shared/pricing-definitions/pricing-definitions-table-data-source.ts

```diff
@@ -0,0 +1,232 @@
+  private context = {
+    entityID: '',
+    entityType: ''
+  };
+  private canCreatePricingDefinition: boolean;
+
+  public constructor(
+    public spinnerService: SpinnerService,
+    public translateService: TranslateService,
+    private messageService: MessageService,
+    private dialogService: DialogService,
+    private router: Router,
+    private dialog: MatDialog,
+    private centralServerService: CentralServerService,
+    private authorizationService: AuthorizationService,
+    private datePipe: AppDatePipe,
+    private appPricingDimensionsUnit: AppPricingDimensionsUnit) {
+    super(spinnerService, translateService);
+    this.canCreatePricingDefinition = this.authorizationService.canCreatePricingDefinition();
+    // Init
+    this.initDataSource();
+  }
+
+  public setContext(entityID: string, entityType: string) {
+    this.context.entityID = entityID;
+    this.context.entityType = entityType;
+  }
+
+  public isContextSet() {
+    return !!(this.context.entityType);
+  }
+
+  public loadDataImpl(): Observable<DataResult<PricingDefinition>> {
+    return new Observable((observer) => {
+      // Get the PricingDefinitions
+      this.centralServerService.getPricingDefinitions(this.buildFilterValues(),
+        this.getPaging(), this.getSorting(), this.context).subscribe((pricingDefinition) => {
+        this.createAction.visible = this.canCreatePricingDefinition;
+        observer.next(pricingDefinition);
+        observer.complete();
+      }, (error) => {
+        // Show error
+        Utils.handleHttpError(error, this.router, this.messageService, this.centralServerService, 'general.error_backend');
+        // Error
+        observer.error(error);
+      });
+    });
+  }
+
+  public buildTableDef(): TableDef {
+    return {
+      search: {
+        enabled: false,
+      },
+      hasDynamicRowAction: true,
+    };
+  }
+
+  public buildTableColumnDefs(): TableColumnDef[] {
+    const columns: TableColumnDef[] = [
+      {
+        id: 'name',
+        name: 'chargers.name',
+        headerClass: 'd-none d-xl-table-cell col-30p',
+        class: 'd-none d-xl-table-cell col-30p',
+        sortable: true,
+      },
+      {
+        id: 'staticRestrictions.validFrom',
+        name: 'settings.pricing.valid_from',
+        formatter: (validFrom: Date) => this.datePipe.transform(validFrom),
+        headerClass: 'd-none d-xl-table-cell col-30p',
+        class: 'd-none d-xl-table-cell col-30p',
+        sortable: true,
+      },
+      {
+        id: 'staticRestrictions.validTo',
+        name: 'settings.pricing.valid_to',
+        formatter: (validTo: Date) => this.datePipe.transform(validTo),
+        headerClass: 'col-15p',
+        class: 'd-none d-xl-table-cell col-30p',
+        sortable: true,
+      },
+      {
+        id: 'dimensions.flatFee.price',
+        name: 'settings.pricing.flat_fee',
+        formatter: (price: number) => {
+          if (price === undefined) {
```

Comment:
Could be part of the appPricingDimensionsUnit to return the '-'?

Reply:
it was to avoid entering appPricingDimensionsUnit pipe when price was undefined

Reply:
will commit change ✅